### PR TITLE
Add initial docs on Python Components

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
Fixes https://github.com/ml6team/fondant/issues/809
Fixes https://github.com/ml6team/fondant/issues/810

### CODE REVIEW

File: docs/components/components.md [hunk 1]
```diff
@@ -26,27 +78,11 @@ We can distinguish two different types of components:
    and write an accompanying component specification that refers to it. This is used for 
   more complex components that require additional dependencies (e.g. GPU support). 
 
+- **Reusable components** can be used out of the box and can be loaded from the fondant 
+  component registry. They are containerized components that are defined by the Fondant team or
```
Comment (mrchtr) [inline docs/components/components.md:82]: nitpick: Would use the term `FondantHub` for consistency.

File: docs/documentation_guide.md [hunk 2]
```diff
@@ -17,8 +17,9 @@ Learn how to use Fondant to build your own data processing pipeline.
 
 -> Design your own fondant [pipeline](pipeline.md) using the Fondant pipeline SDK.  
 -> Use existing [reusable components](components/hub.md) to build your pipeline.
--> Build your own [custom component](components/custom_component.md) using the Fondant component
-SDK.  
+-> Build your own custom [python component](components/custom_containerized_component.md) 
+or [containerized component](components/custom_containerized_component.md) using the Fondant component
```
Comment (mrchtr) [inline docs/documentation_guide.md:21]: Maybe we can use something like `share components`. Then the story would be, build your custom component with the python component and later on, if you want to share/publish it, the more "complex" documentation how to package the component into a reusable one.

Comment (PhilippeMoussalli) [reply docs/documentation_guide.md:21]: This makes sense!

**Review by mrchtr — APPROVED** [review]
Thanks @PhilippeMoussalli!

**Pull request merged**